homeassistant/components/shelly/light.py

```diff
@@ -93,33 +151,40 @@ def color_temp(self) -> Optional[float]:
         # If you set DUO to max mireds in Shelly app, 2700K,
         # It reports 0 temp
         if color_temp == 0:
-            return self.max_mireds
+            return min_kelvin(self.wrapper.model)
 
         return int(color_temperature_kelvin_to_mired(color_temp))
 
     @property
-    def min_mireds(self) -> float:
+    def min_mireds(self) -> Optional[float]:
         """Return the coldest color_temp that this light supports."""
-        return color_temperature_kelvin_to_mired(6500)
+        return color_temperature_kelvin_to_mired(KELVIN_MAX_VALUE)
 
     @property
-    def max_mireds(self) -> float:
+    def max_mireds(self) -> Optional[float]:
         """Return the warmest color_temp that this light supports."""
-        return color_temperature_kelvin_to_mired(2700)
+        return color_temperature_kelvin_to_mired(min_kelvin(self.wrapper.model))
 
     async def async_turn_on(self, **kwargs) -> None:
         """Turn on light."""
         params = {"turn": "on"}
         if ATTR_BRIGHTNESS in kwargs:
             tmp_brightness = kwargs[ATTR_BRIGHTNESS]
-            params["brightness"] = int(tmp_brightness / 255 * 100)
+            params["brightness"] = params["gain"] = int(tmp_brightness / 255 * 100)
         if ATTR_COLOR_TEMP in kwargs:
             color_temp = color_temperature_mired_to_kelvin(kwargs[ATTR_COLOR_TEMP])
-            if color_temp > 6500:
-                color_temp = 6500
-            elif color_temp < 2700:
-                color_temp = 2700
+            if color_temp > KELVIN_MAX_VALUE:
+                color_temp = KELVIN_MAX_VALUE
+            elif color_temp < min_kelvin(self.wrapper.model):
+                color_temp = min_kelvin(self.wrapper.model)
             params["temp"] = int(color_temp)
+        if ATTR_HS_COLOR in kwargs:
```

Comment:
This should be an `elif`. We should either send CT or RRGB or White value.
```suggestion
        elif ATTR_HS_COLOR in kwargs:
```